clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -218,10 +218,10 @@ public synchronized boolean updateRequested() {
     }
 
     /**
-     * @return the topic ID for the given topic name or null if the ID does not exist or is not known
+     * @return a mapping from topic names to topic IDs for all topics with valid IDs in the cache
      */
-    public synchronized Uuid topicId(String topicName) {
```

Comment:
Any harm keeping this one? Seems like it simplified some of the uses, especially in tests.

Reply:
Hmmm...this is the version that could have a new cache value? Only thing I might worry about is misuse.